neuralforecast/core.py

```diff
@@ -464,19 +492,40 @@ def fit(
                 target_col=target_col,
                 distributed_config=distributed_config,
             )
+        elif isinstance(df, List):
+            if not all(isinstance(val, str) for val in df):
+                raise ValueError(
+                    "All entries in the list of files must be of type string"
+                )
+            self.dataset = self._prepare_iterative_fit(
+                df=df,
+                static_df=static_df,
+                sort_df=sort_df,
+                id_col=id_col,
+                time_col=time_col,
+                target_col=target_col,
+            )
+            self.uids = self.dataset.indices
+            self.last_dates = self.dataset.last_times
         elif df is None:
             if verbose:
                 print("Using stored dataset.")
         else:
             raise ValueError(
-                f"`df` must be a pandas, polars or spark DataFrame or `None`, got: {type(df)}"
+                f"`df` must be a list of parquet files, or a pandas, polars or spark DataFrame or `None`, got: {type(df)}"
             )
 
         if val_size is not None:
-            if self.dataset.min_size < val_size:
-                warnings.warn(
-                    "Validation set size is larger than the shorter time-series."
-                )
+            if isinstance(self.dataset, IterativeTimeSeriesDataset):
```

Comment:
would be better if min_size was set on `IterativeTimeSeriesDataset` so that we can do this check directly. Ideally both `IterativeTimeSeriesDataset` and `TimeseriesDataset` would inheret from some base class that have all the same properties and methods and avoid any if `isinstance()` checks. Maybe the only difference would be the `from_dataset` vs `from_df` method between them?

Reply:
I've added a BaseTimeSeriesDataset for this now